src/pybind/mgr/dashboard/module.py

```diff
@@ -360,28 +366,26 @@ def shutdown(self):
     @CLIWriteCommand("dashboard set-ssl-certificate")
     def set_ssl_certificate(self,
                             mgr_id: Optional[str] = None,
-                            inbuf: Optional[bytes] = None):
+                            inbuf: Optional[str] = None):
         if inbuf is None:
-            return -errno.EINVAL, '',\
-                'Please specify the certificate file with "-i" option'
+            return -errno.EINVAL, '', self.MSG_SSL_CERT_ERROR
         if mgr_id is not None:
-            self.set_store(_get_localized_key(mgr_id, 'crt'), inbuf.decode())
+            self.set_store(_get_localized_key(mgr_id, 'crt'), inbuf)
         else:
-            self.set_store('crt', inbuf.decode())
-        return 0, 'SSL certificate updated', ''
+            self.set_store('crt', inbuf)
+        return 0, self.MSG_SSL_CERT_UPDATED, ''
 
     @CLIWriteCommand("dashboard set-ssl-certificate-key")
     def set_ssl_certificate_key(self,
                                 mgr_id: Optional[str] = None,
-                                inbuf: Optional[bytes] = None):
+                                inbuf: Optional[str] = None):
         if inbuf is None:
-            return -errno.EINVAL, '',\
-                'Please specify the certificate key file with "-i" option'
+            return -errno.EINVAL, '', self.MSG_SSL_CERT_KEY_ERROR
         if mgr_id is not None:
-            self.set_store(_get_localized_key(mgr_id, 'key'), inbuf.decode())
+            self.set_store(_get_localized_key(mgr_id, 'key'), inbuf)
         else:
-            self.set_store('key', inbuf.decode())
-        return 0, 'SSL certificate key updated', ''
+            self.set_store('key', inbuf)
+        return 0, self.MSG_SSL_CERT_KEY_UPDATED, ''
```

Comment:
I see these 2 (or 3 actually) functions are exactly the same except for some part in the error message (`certificate file` vs `certificate key`) and the key name... So, what about a parameterized function:
```py
# Literal may need typing_extensions in 3.6
def set_ssl_item(self, item_label: str, item_key: Literal['crt', 'key'], mgr_id: Optional[str]):
  if inbuf is None:
    return -errno.EINVAL, '', 'Please specify the {} with "-i" option'.format(item_label)

  if mgr_id is not None:
    self.set_store(_get_localized_key(mgr_id, item_key), inbuf)
  else:
    self.set_store(item_key, inbuf)
  return 0, 'SSL {item} updated'.format(item_label), ''

@CLIWriteCommand("dashboard set-ssl-certificate")
def set_ssl_certificate(self, mgr_id: Optional[str] = None, inbuf: Optional[str] = None):
  return self.set_ssl_item('certificate', 'crt', mgr_id)

@CLIWriteCommand("dashboard set-ssl-certificate-key")
def set_ssl_certificate(self, mgr_id: Optional[str] = None, inbuf: Optional[str] = None):
  return self.set_ssl_item('certificate key', 'key', mgr_id)

@CLIWriteCommand("dashboard create-self-signed-cert")
def create_self_signed_cert(self):
  cert, pkey = create_self_signed_cert('IT', 'ceph-dashboard')
  r = HandleCommandResult(*self.set_ssl_item('self-signed certificate', 'crt', mgr_id=None))
  if r.retval != 0:
    return r

  r = HandleCommandResult(*self.set_ssl_item('self-signed certificate key', 'key', mgr_id=None))
  if r.retval != 0:
    return r
  return 0, Self-signed certificate created', ''
```

Reply:
The more I see these functions (which require the Dashboard to be running) the more I'm convinced that we should remove all the dummy commands that can be easily replaced with a `ceph config set...` (which is something that can be run before starting the dashboard) :/

Reply:
> I see these 2 (or 3 actually) functions are exactly the same except for some part in [...]

Of course more refactoring can be done, but as per our previous agreement in Dashboard Team regarding bugfixes,
my intention here was to provide only the minimal fix covered by tests.